scripts/upload_results_to_oep.py

```diff
@@ -0,0 +1,247 @@
+# coding: utf-8
+r"""
+Inputs
+-------
+in_path: str
+    ``results/{scenario}/b3_results/data``: path to scalar results in oemof-B3-format
+out_path: str
+    ``results/{scenario}/b3_results/metadata``: path to metadata directory
+params:
+    ``results/{scenario}/{scenario}.log``: path to logfile
+
+Outputs
+---------
+``results/{scenario}/b3_results/metadata``: directory with automatically created metadata.
+
+Description
+-------------
+This script performs the following steps:
+
+* Create metadata (in oemetadata format, using the template in ``oemof-B3/schema/oemetadata.json``)
+  for given model results (in oemof-B3-format) and saves them locally as json files.
+* Create tables on the OpenEnergyPlatform (OEP) based on the metadata.
+* Upload model results to the OEP tables.
+* Validate metadata.
+* Upload metadata to the OEP tables.
+
+The script can delete existing tables on the OEP that have been uploaded with an affiliated OEP user
+account.
+
+The oemetadata format is a standardised json file format and is required for all data uploaded to
+the OEP. It includes the data model, the used data types, and general information about the data
+context. Tables in sqlalchemy are created based on the information in the oemetadata.
+"""
+import json
+import logging
+import os
+import pathlib
+import sys
+from datetime import date
+
+import pandas as pd
+
+from oemof_b3.config import config
+from oemof_b3.schema import oemetadata_scal, oemetadata_ts, SCHEMA_SCAL, SCHEMA_TS
+from oemof_b3.tools import data_processing as dp
+
+try:
+    from oem2orm import oep_oedialect_oem2orm as oem2orm
+except ImportError:
+    raise ImportError("Need to install oem2orm to upload results to OEP.")
+
+logger = logging.getLogger()
+
+# try to get oep_user and oep_token from .secrets.yaml
+try:
+    os.environ["OEP_USER"] = config.settings.oep_user
+    os.environ["OEP_TOKEN"] = config.settings.oep_token
+except AttributeError:
+    logger.warning(
+        "No oep_user and/or oep_token provided in oemof_b3/config/.secrets.yaml. "
+        "Will have to type provide them manually when uploading to OEP."
+    )
+
+
+SCHEMA = "model_draft"
```

Comment:
JFI: you can get the schema from the metadata -> in the field resources - name: there should always be a value like "schema.tablename" but it might be saver to use this variable.

Reply:
Not sure if I understand you - I expected the schema to be undefined in the metadata template so that the user can set it?

Reply:
True :) I take back my comment!